src/ui-components/Address.tsx

```diff
@@ -71,6 +73,13 @@ const Address = ({ address, className, displayInline, disableIdenticon, extensio
 		}
 	};
 
+	const getKiltName = async () => {
+		const web3Name = await getKiltDidName(address)
+		if(web3Name) {
+			setKiltName(web3Name!)
```

Comment:
instead of enforcing with `!`, please use web3Name `setKiltName(web3Name || '')`.

The whole if block can be removed after the above.